### DISCUSSION TITLE
mysql mapping to risingwave data types, such as smallint, bigint, datetime, timestamp, text, blob,

### DISCUSSION THREAD

**Discussion by findpather** [risingwavelabs/risingwave#10129] [General]
mysql corresponds to risingwave data types, such as smallint, bigint, datetime, timestamp, text, blob, etc
Do we have a comprehensive correspondence here?

data structure like the following, How do I configure it reasonably in risingwave?
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `journal_id` int(10) unsigned NOT NULL,
  `issue_id` int(10) unsigned NOT NULL,
  `section_id` int(10) unsigned DEFAULT NULL,
  `special_issue_id` int(10) unsigned DEFAULT NULL,
  `article_type_id` int(10) unsigned NOT NULL,
  `publisher` varchar(70) COLLATE utf8_unicode_ci DEFAULT NULL,
  `volume` smallint(4) unsigned NOT NULL,
  `year` smallint(4) unsigned DEFAULT NULL,
  `issue` tinyint(3) unsigned NOT NULL,
  `number` varchar(6) COLLATE utf8_unicode_ci DEFAULT NULL,
  `firstpage` mediumint(6) unsigned DEFAULT NULL,
  `lastpage` mediumint(6) unsigned DEFAULT NULL,
  `pubdate_published` date NOT NULL,
  `pubdate_received` date NOT NULL,
  `pubdate_revised` date NOT NULL,
  `pubdate_accepted` date NOT NULL,
  `language` varchar(2) COLLATE utf8_unicode_ci NOT NULL,
  `title` mediumtext COLLATE utf8_unicode_ci NOT NULL,
  `titlenote` text COLLATE utf8_unicode_ci,
  `abstract` longtext COLLATE utf8_unicode_ci NOT NULL,
  `keywords` text COLLATE utf8_unicode_ci NOT NULL,
  `summary` mediumtext COLLATE utf8_unicode_ci,
  `application` text COLLATE utf8_unicode_ci,
  `id_pii` varchar(25) COLLATE utf8_unicode_ci NOT NULL COMMENT 'Publisher Identifier Information',
  `id_doi` varchar(50) COLLATE utf8_unicode_ci NOT NULL COMMENT 'Digital Object Identifier ID',
  `id_doi_batch` varchar(25) COLLATE utf8_unicode_ci DEFAULT NULL,
  `id_doi_submitted` tinyint(1) NOT NULL DEFAULT '0',
  `id_pmid` varchar(10) COLLATE utf8_unicode_ci DEFAULT NULL,
  `submission_manuscript_id` int(10) unsigned DEFAULT NULL,
  `manuscript_id` varchar(40) COLLATE utf8_unicode_ci DEFAULT NULL,
  `editors_pick` tinyint(1) DEFAULT NULL,
  `frozen` tinyint(1) unsigned NOT NULL DEFAULT '0',
  `date_inserted` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
  `date_updated` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `feature_paper` tinyint(1) NOT NULL DEFAULT '0',
  `reprints_ordered` tinyint(1) NOT NULL DEFAULT '0',
  `reference_batch_id` varchar(50) COLLATE utf8_unicode_ci DEFAULT NULL,
  `reference_submitted` tinyint(1) NOT NULL DEFAULT '0',
  `reference_citedby_number` int(4) DEFAULT NULL,
  `check_sum` varchar(32) COLLATE utf8_unicode_ci DEFAULT NULL,
  `citations_to_update` tinyint(1) DEFAULT '0',
  `citations_last_update` timestamp NULL DEFAULT NULL,
  `is_public` tinyint(1) NOT NULL DEFAULT '0',
  `doi_checker_indicator` tinyint(1) NOT NULL DEFAULT '0',
  `is_xml_parsing` tinyint(1) DEFAULT '0',
  `is_notable` tinyint(1) DEFAULT '0',
  `scitations_done` tinyint(1) DEFAULT '0',
  `article_copyrights_id` int(11) DEFAULT '1',
  `conflict` text COLLATE utf8_unicode_ci,
  `editors_choice` tinyint(1) DEFAULT '0',
  `order_num` int(10) DEFAULT NULL,
  `files_update` datetime DEFAULT NULL,
  `files_archive` datetime DEFAULT NULL,
  `reference_citedby_number_max` int(11) DEFAULT NULL,
  `is_roman_numbers` tinyint(1) DEFAULT '0',
  `book_builder_ready` tinyint(4) NOT NULL DEFAULT '0',
  `multidisciplinary_topic_id` int(10) unsigned DEFAULT NULL,
  `latex_version` int(11) DEFAULT NULL,
  `date_published` datetime DEFAULT NULL,
  `latex_version_last_compilation` datetime DEFAULT NULL,
  `nb_pre_compilations` int(11) DEFAULT '0',
  PRIMARY KEY (`id`),
  UNIQUE KEY `articles_un_id_doi` (`id_doi`),

**Comment by neverchanje** [comment]
Just curious...May I know which MySQL version are you using? I encountered "ERROR 1067 (42000): Invalid default value for 'date_inserted'" in mysql 5.7 when I tried to create the table you provided.
